test/document.test.js

```diff
@@ -11551,6 +11551,202 @@ describe('document', function() {
     assert.ok(err);
     assert.ok(err.errors['testProp.testSubProp.nested.from']);
   });
+
+  describe('$inc (gh-11915)', function() {
+    describe('top-level path', function() {
+      let Test;
+
+      beforeEach(function() {
+        const schema = new Schema({
+          counter: Number
+        });
+        Test = db.model('Test', schema);
+      });
+
+      it('sends a $inc command for a given path', async function() {
+        await Test.create({ counter: 0 });
+        const doc = await Test.findOne();
+        assert.strictEqual(doc.counter, 0);
+        const doc2 = await Test.findOne();
+
+        doc2.counter = 1;
+        await doc2.save();
+
+        doc.$inc('counter');
```

Comment:
Would be nice to have the test explicitly increment specific value, to have some idea how it works when looking at the tests as an example:
```suggestion
        doc.$inc('counter', 1);
```